torch/_inductor/graph.py

```diff
@@ -504,9 +674,16 @@ def run_node(self, n: torch.fx.Node):
                 # requiring a stride order for a non-dense output wouldn't
                 # recreate the same strides, and would fail with view, defer for now.
                 if dense and len(strides):
-                    result = ir.ExternKernel.require_stride_order(
-                        result, ir.get_stride_order(strides)
-                    )
+                    stride_order = ir.get_stride_order(strides)
+
+                    if (
```

Comment:
Can we change this logic to 
```
if n.name in self.user_visible_outputs:
     result = ir.ExternKernel.require_stride_order(result, stride_order)
```
?

Reply:
I created a temporary PR to test this out: https://github.com/pytorch/pytorch/pull/102783 

Let's see how the perf looks

Reply:
The change I tried in https://github.com/pytorch/pytorch/pull/102783 has some perf hit. E.g. res2net101_26w_4s slows down from 1.158x to 1.057x on my dev environment. I think the reason is `pick_loop_order` may not as accurate (or as specific) as `self.nodes_prefer_channels_last` to force a tensor to be channels last.

I could move 
```
                    if (
                        len(result.get_size()) == 4
                        and n in self.nodes_prefer_channels_last
                        and n.name not in self.user_visible_outputs
                    ):
                        stride_order = ir.NHWC_STRIDE_ORDER
```         
to finalize, but in finalize we decide a list of buffer's layout rather than nodes. We would need some way to map buffer back to node (so we can check against `self.nodes_prefer_channels_last`). Maybe we can use IRNode.origins to do that. But I'm not sure if it's reliable.